flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -246,13 +252,16 @@ public static TaskManagerServices fromConfiguration(
 		// start the I/O manager, it will create some temp directories.
 		final IOManager ioManager = new IOManagerAsync(taskManagerServicesConfiguration.getTmpDirPaths());
 
-		final NetworkEnvironment network = NetworkEnvironment.create(
+		final ShuffleEnvironment shuffleEnvironment = NettyShuffleEnvironment.fromConfiguration(
```

Comment:
Can't we keep the `NettyShuffleEnvironmentConfiguration` part of the `TaskManagerServicesConfiguration`? I think in the future we will need a `ShuffleEnvironmentConfiguration` which we pass to the `ShuffleEnvironmentFactory` once the `ShuffleEnvironment` is pluggable.

Reply:
`NettyShuffleEnvironmentConfiguration` is netty specific. I will add `ShuffleEnvironmentContext` to generate it from `TaskManagerServicesConfiguration` and pass to the factory.